tests/js/spec/views/dashboardsV2/widgetBuilder/widgetBuilder.spec.tsx

```diff
@@ -1,63 +1,176 @@
 import React from 'react';
+import {urlEncode} from '@sentry/utils';
 
 import {initializeOrg} from 'sentry-test/initializeOrg';
-import {mountWithTheme, screen} from 'sentry-test/reactTestingLibrary';
+import {
+  mountWithTheme,
+  screen,
+  userEvent,
+  waitFor,
+} from 'sentry-test/reactTestingLibrary';
+import {textWithMarkupMatcher} from 'sentry-test/utils';
 
 import {
   DashboardDetails,
   DashboardWidgetSource,
   DisplayType,
   Widget,
 } from 'sentry/views/dashboardsV2/types';
-import WidgetBuilder from 'sentry/views/dashboardsV2/widgetBuilder';
+import * as dashboardsTypes from 'sentry/views/dashboardsV2/types';
+import WidgetBuilder, {WidgetBuilderProps} from 'sentry/views/dashboardsV2/widgetBuilder';
+
+function renderTestComponent({
+  widget,
+  dashboard,
+  query,
+  orgFeatures,
+  onSave,
+}: {
+  dashboard?: WidgetBuilderProps['dashboard'];
+  onSave?: WidgetBuilderProps['onSave'];
+  orgFeatures?: string[];
+  query?: Record<string, any>;
+  widget?: WidgetBuilderProps['widget'];
+} = {}) {
+  const {organization, router, routerContext} = initializeOrg({
+    ...initializeOrg(),
+    organization: {
+      features: orgFeatures ?? [
+        'new-widget-builder-experience',
+        'dashboards-edit',
+        'global-views',
+      ],
+    },
+    router: {
+      location: {
+        query: {
+          source: DashboardWidgetSource.DASHBOARDS,
+          ...(query ?? {}),
+        },
+      },
+    },
+  });
+
+  mountWithTheme(
+    <WidgetBuilder
+      route={{}}
+      router={router}
+      routes={router.routes}
+      routeParams={router.params}
+      location={router.location}
+      dashboard={
+        dashboard ?? {
+          id: '1',
+          title: 'Dashboard',
+          createdBy: undefined,
+          dateCreated: '2020-01-01T00:00:00.000Z',
+          widgets: [],
+        }
+      }
+      onSave={onSave ?? jest.fn()}
+      widget={widget}
+      params={{
+        orgId: organization.slug,
+        widgetIndex: widget ? Number(widget.id) : undefined,
+      }}
+    />,
+    {
+      context: routerContext,
+      organization,
+    }
+  );
+
+  return {router};
```

Comment:
not really... the only thing the tests currently need is the `router`. I would return mountWithTheme only in case we need the container, so we could use for snapshot testing

Reply:
Right, makes sense after seeing the other tests below! Thanks for the clarification